### PR TITLE
chore!: RFC9420, update deps & UniFFI 0.24 upgrade [WPB-2785] [WPB-1180] [WPB-579]

### PR DESCRIPTION
**Pull request by OtaK**
Also switches to RFC9420

Closes #334 
Closes #335 
Closes #343 
Closes #344 
Closes #340

### CODE REVIEW

File: crypto-ffi/Cargo.toml [hunk 1]
```diff
@@ -9,12 +9,16 @@ publish = false
 
 [lib]
 name = "core_crypto_ffi"
-crate-type = ["cdylib", "staticlib"]
```
Comment (beltram) [inline crypto-ffi/Cargo.toml:12]: Won't this bloat the binary like the last time we tried ?

File: crypto-ffi/build.rs [hunk 2]
```diff
@@ -19,35 +19,35 @@ const UDL_FILE: &str = "./src/CoreCrypto.udl";
...
-                Some("./bindings/swift/WireCoreCrypto/WireCoreCrypto".into()),
-                None,
-                false,
-            )
-            .unwrap();
-        } else if #[cfg(target_family = "wasm")] {
-            println!("cargo:rustc-cfg=wasm");
+            uniffi::generate_scaffolding(UDL_FILE).unwrap();
+
+            // uniffi::generate_bindings(
```
Comment (beltram) [inline crypto-ffi/build.rs:32]: delete commented out code ?

File: interop/Cargo.toml [hunk 3]
```diff
@@ -5,7 +5,9 @@ edition = "2021"
 license = "GPL-3.0-only"
 
 [features]
-default = ["proteus"]
+default = []
```
Comment (beltram) [inline interop/Cargo.toml:8]: having the proteus feature not default is a good thing (for local tests) but you should force it in CI jobs i.e. "cargo run --bin interop --features proteus"

File: interop/src/clients/corecrypto/ffi.rs [hunk 4]
```diff
@@ -33,7 +33,7 @@ pub struct CoreCryptoFfiClient<'a> {
 impl<'a> CoreCryptoFfiClient<'a> {
     pub async fn new() -> Result<CoreCryptoFfiClient<'a>> {
         let client_id = uuid::Uuid::new_v4();
-        let ciphersuite = CiphersuiteName::MLS_128_DHKEMX25519_AES128GCM_SHA256_Ed25519;
+        let ciphersuite = CiphersuiteName::MLS_128_X25519KYBER768DRAFT00_AES128GCM_SHA256_Ed25519;
```
Comment (beltram) [inline interop/src/clients/corecrypto/ffi.rs:36]: You should probably rollback this to the x25519 ciphersuite

File: interop/src/main.rs [hunk 5]
```diff
@@ -202,24 +217,39 @@ async fn run_mls_test(chrome_driver_addr: &std::net::SocketAddr) -> Result<()> {
...
             .decrypt_message(&conversation_id, message_to_decrypt)
             .await?
             .app_msg
             .ok_or_else(|| eyre!("[MLS] No message recieved on master client"))?;
 
+        let decrypted_master = String::from_utf8(decrypted_master_raw)?;
+
         assert_eq!(decrypted_master, message);
 
+        // tokio::time::sleep(std::time::Duration::from_millis(200)).await;
```
Comment (beltram) [inline interop/src/main.rs:251]: remove commented out code

File: keystore/src/connection/mod.rs [hunk 6]
```diff
@@ -43,8 +42,14 @@ use std::sync::Arc;
 /// See: [IndexedDB limits](https://stackoverflow.com/a/63019999/1934177)
 pub const MAX_BLOB_LEN: usize = 1_000_000_000;
 
-#[async_trait(?Send)]
-pub trait DatabaseConnection: Sized {
+#[cfg(not(target_family = "wasm"))]
+pub trait DatabaseConnectionRequirements: Sized + Send {}
```
Comment (beltram) [inline keystore/src/connection/mod.rs:46]: Maybe add a comment to explain why this is required (uniffi)

**Review by beltram — APPROVED** [review]

File: crypto-ffi/Cargo.toml (hunk 1 above)
Comment (OtaK) [reply crypto-ffi/Cargo.toml:12]: You're right, I'll revert the change

File: interop/Cargo.toml (hunk 3 above)
Comment (OtaK) [reply interop/Cargo.toml:8]: I think it's an artifact of running things fast. I'll revert it, it has no reason to be changed in this PR

**Pull request merged**